client/lib/posts/post-edit-store.js

```diff
@@ -471,7 +471,23 @@ PostEditStore = {
 	},
 
 	isDirty: function() {
-		return _post !== _savedPost || _rawContent !== _initialRawContent;
+		debug(
+			'isDirty post?%s rawContent?%s initial=%s content=%s',
+			_post !== _savedPost,
+			_rawContent !== _initialRawContent,
+			_initialRawContent,
+			_rawContent
+		);
+		if ( _post !== _savedPost ) {
+			return true;
+		}
+		if ( _rawContent !== _initialRawContent ) {
+			return (
+				! isContentEmpty( _rawContent ) ||
```

Comment:
I tried this out, but it appears that TinyMCE is fighting with this content replacement.  When switching between Visual and HTML tabs, I see the content ending up as `<br />`, which is not currently considered "empty", so the post is still marked dirty.  Also, the replacement unexpectedly clears the post content when you enter a blank line as the only content.